### ISSUE TITLE
Zeitkit.restnetworking.apierror error2

### ISSUE THREAD

**Issue by fkaCraig** [alexanderwe/Zeit#2]
Just downloaded for the first time, logged in and I'm getting this error.

![image](https://user-images.githubusercontent.com/40214242/82840752-9e52d600-9ecb-11ea-8da1-705a23ac8c67.png)

**Comment by fkaCraig** [comment]
![image](https://user-images.githubusercontent.com/40214242/82840724-8ed38d00-9ecb-11ea-8dda-08a2e80f7096.png)

**Comment by PaddyTB** [comment]
I too have this issue after authorising the app with https://web.timingapp.com/

**Comment by alexanderwe** [comment]
@fkaCraig @PaddyTB thanks for reporting this issue. Since I do not track anything in the application it would be nice if you can answer me some questions: 

- Does the error appear even after completely closing the application and re-opening it again or only the time after successfully logging in ? 
- Are there any special color values for your projects or any lengthy names you are using for your projects/time tracking entries  ? 

Thanks for your help. I will definitely work on some better error messages in the next version to make debugging more easier.

**Comment by PaddyTB** [comment]
Hi Alexander,

The error persists even after completely shutting down the app and restarting it.

I’m not sure what you would call a special colour value - IIRC they were just assigned from the standard pickiist but I can’t guarantee that.

Regarding project names, what number of characters would you call lengthy?

The app looks great so far but we haven’t got to see any of the juicy bits yet!

Cheers,
Paddy

On 26 May 2020, at 06:44, Alexander Weiß <notifications@github.com> wrote:

﻿
@fkaCraig @PaddyTB thanks for reporting this issue. Since I do not track anything in the application it would be nice if you can answer me some questions:

Does the error appear even after completely closing the application and re-opening it again or only the time after successfully logging in ?
Are there any special color values for your projects or any lengthy names you are using for your projects/time tracking entries ?
Thanks for your help. I will definitely work on some better error messages in the next version to make debugging more easier.

**Comment by PaddyTB** [comment]
I also deleted and reinstalled the app having deauthorised it in the TimingApp website.

HTH,
Paddy

On 26 May 2020, at 08:05, Paddy Brown <paddy.brown@tanguy.co.uk> wrote:

﻿Hi Alexander,

The error persists even after completely shutting down the app and restarting it.

I’m not sure what you would call a special colour value - IIRC they were just assigned from the standard pickiist but I can’t guarantee that.

Regarding project names, what number of characters would you call lengthy?

The app looks great so far but we haven’t got to see any of the juicy bits yet!

Cheers,
Paddy

On 26 May 2020, at 06:44, Alexander Weiß <notifications@github.com> wrote:

﻿
@fkaCraig @PaddyTB thanks for reporting this issue. Since I do not track anything in the application it would be nice if you can answer me some questions:

Does the error appear even after completely closing the application and re-opening it again or only the time after successfully logging in ?
Are there any special color values for your projects or any lengthy names you are using for your projects/time tracking entries ?
Thanks for your help. I will definitely work on some better error messages in the next version to make debugging more easier.

**Comment by alexanderwe** [comment]
@PaddyTB Thanks for your response. I was just searching for some reasons why the app is crashing. Currently I am assuming there is some data that cannot be correctly deserialized and therefore the app is showing the error. 

It could be that there is some edge case in your data that I am not aware of. I will try to implement a better error message until the end of the upcoming weekend and then provide a new beta version to TestFlight. Maybe you can then try that updated version and see if the error persists and maybe we can fix the issue with the updated error messages. 

I didn’t want to implement some fancy error reporting tools yet, which would give me error messages and stack traces while the app is running, because I didn’t want to track anything from any user of the app. That’s why debugging such an issue you two currently have is a little bit time consuming. I am sorry that inconvenience. Maybe I should think about implementing such a tool. Would you as a user be fine with such tools ?

**Comment by fkaCraig** [comment]
Hi

No this just started like this. I’ve never actually been able to use it.

**Comment by PaddyTB** [comment]
Happy to help - I guess any error report should be fine provided the user can review what is being sent to and exclude anything that might be sensitive - that’s how the standard Apple reporting works as I’m sure you know.

Let me know what other debugging steps you want and I’ll happily follow them.

Paddy

On 26 May 2020, at 09:21, Alexander Weiß <notifications@github.com> wrote:


@PaddyTB <https://github.com/PaddyTB> Thanks for your response. I was just searching for some reasons why the app is crashing. Currently I am assuming there is some data that cannot be correctly deserialized and therefore the app is showing the error.

It could be that there is some edge case in your data that I am not aware of. I will try to implement a better error message until the end of the upcoming weekend and then provide a new beta version to TestFlight. Maybe you can then try that updated version and see if the error persists and maybe we can fix the issue with the updated error messages.

I didn’t want to implement some fancy error reporting tools yet, which would give me error messages and stack traces while the app is running, because I didn’t want to track anything from any user of the app. That’s why debugging such an issue you two currently have is a little bit time consuming. I am sorry that inconvenience. Maybe I should think about implementing such a tool. Would you as a user be fine with such tools ?

**Comment by MrMage** [comment]
Daniel here, the developer of Timing. I might have been able to fix this issue on the Timing API back-end now. Could you please try again to see if the errors persist?

**Comment by PaddyTB** [comment]
Yes, error is gone, thanks, I’ll try to do some real tasks with the app now.

Paddy

On 27 May 2020, at 09:02, Daniel Alm <notifications@github.com> wrote:

﻿
Daniel here, the developer of Timing. I might have been able to fix this issue on the Timing API back-end now. Could you please try again to see if the errors persist?

**Comment by alexanderwe** [comment]
@MrMage Thanks for jumping in an fixing the issue server side !  
@PaddyTB glad to hear that the app is working for you now.
@fkaCraig is the error gone for you as well ?

**Comment by maparent** [comment]
I also had that error and saw it solved.

**Comment by fkaCraig** [comment]
No. Still getting the same problem. Tried closing the app down. Restart etc. Still the same.

**Comment by MrMage** [comment]
Craig, which edition of Timing are you using? Timing Web (and thus Zeit) can not be used with Timing Productivity.

**Comment by alexanderwe** [comment]
@fkaCraig A new api error message view was implemented in v1.10.  Do you still have that issue ?
